### PR TITLE
fix: code refactor based on review comments

### PR DESCRIPTION
**Pull request by gongyj-ibm**

### CODE REVIEW

**Review by iamniting — CHANGES_REQUESTED** [review]
Can we separate changes into multiple commits. each for a different review or comment.

**Comment by gongyj-ibm** [comment]
> Can we separate changes into multiple commits. each for a different review or comment.

done, thanks.

File: controllers/flashsystemcluster_controller.go [hunk 1]
```diff
@@ -183,18 +183,12 @@ func (r *FlashSystemClusterReconciler) Reconcile(ctx context.Context, req ctrl.R
...
-	r.Log.Info("step: init status/conditions of the FlashSystemCluster resource")
-	if instance.Status.Conditions == nil {
+	r.Log.Info("step: reset progressing conditions of the FlashSystemCluster resource")
+	if util.IsStatusConditionFalse(instance.Status.Conditions, odfv1alpha1.ConditionProgressing) {
 		reason := odfv1alpha1.ReasonReconcileInit
-		message := "Initializing flashsystem ODF resources"
+		message := "processing flashsystem ODF resources"
 		util.SetReconcileProgressingCondition(&instance.Status.Conditions, reason, message)
 		instance.Status.Phase = util.PhaseProgressing
-
```
Comment (iamniting) [inline controllers/flashsystemcluster_controller.go:192]: why did you remove the update status call?

File: controllers/flashsystemcluster_controller.go [hunk 2]
```diff
@@ -509,41 +509,35 @@ func (r *FlashSystemClusterReconciler) ensureDefaultStorageClass(instance *odfv1
 
 	err := r.Client.Get(
 		context.TODO(),
-		types.NamespacedName{Name: instance.Spec.DefaultPool.StorageClassName, Namespace: ""},
+		types.NamespacedName{Name: instance.Spec.DefaultPool.StorageClassName},
 		foundStorageClass)
 	if err != nil {
```
Comment (iamniting) [inline controllers/flashsystemcluster_controller.go:514]: you can rephrase this in such a way that you can have create call only once.

if found {
  check if it is as expected, if yes return from here, if not delete it
}

create a storage class

**Review by iamniting — CHANGES_REQUESTED** [review]
Screenshot from 2021-07-14 14-57-06

typo should be fixed in commit.

File: controllers/flashsystemcluster_controller.go (hunk 2 above)
Comment (gongyj-ibm) [reply controllers/flashsystemcluster_controller.go:514]: yeah, good catch.

Comment (gongyj-ibm) [reply controllers/flashsystemcluster_controller.go:516]: done, thanks.

**Comment by gongyj-ibm** [comment]
> Screenshot from 2021-07-14 14-57-06
> 
> typo should be fixed in commit.

done

File: controllers/flashsystemcluster_controller.go (hunk 1 above)
Comment (gongyj-ibm) [reply controllers/flashsystemcluster_controller.go:192]: Previous update status call is used for condition initialization which only executes once when CR is created.
Now the `progressing` is set for each round of reconciling, we could leverage status update to only update once at the end of reconcile function in both good path and any error handling.

**Review by iamniting — APPROVED** [review]
